src/compat_win32.cpp

```diff
@@ -32,14 +32,18 @@ bool unc_getenv(const char *name, std::string &str)
    if (buf)
    {
       len = GetEnvironmentVariableA(name, buf, len);
-   }
-   buf[len] = 0;
+      buf[len] = 0;
 
-   str = buf;
-   //printf("%s: name=%s len=%zu value=%s\n", __func__, name, len, str.c_str());
-   free(buf);
+      str = buf;
+      //printf("%s: name=%s len=%zu value=%s\n", __func__, name, len, str.c_str());
+      free(buf);
 
-   return(true);
+      return(true);
+   }
+   else
+   {
+      return(false);
```

Comment:
we should make sure 'str' is an empty string in this case, since now it would not be updated if we go through the else part.